snek/snek.py

```diff
@@ -251,20 +251,23 @@ def __load_sitemap(self):
                 'date': None
             }
 
-            # Load from front-matter and add to default
-            try:
-                metadata_from_file = frontmatter.load(filepath, handler=frontmatter_json_handler()).metadata
-                for key, value in metadata_from_file.items():
-                    metadata[key] = value
-
             # If the file could not be read or open, go to next file
-            except FileNotFoundError:
+            if os.access(filepath, os.R_OK):
                 self.__add_error(f"{filepath} cannot be read.")
                 continue
+
+            # Load from front-matter and add to default
+            try:
+                metadata_from_file = frontmatter.load(filepath, handler=frontmatter_json_handler()).metadata
             # If the file's content is not valid JSON frontmatter, simply log it
             except json.decoder.JSONDecodeError as err:
                 self.__add_error(f"{filepath} does not contain valid JSON. {err}")
 
+            # We don't want `filepath` from metadata to change our dict
+            metadata.pop("filepath", None)
+            
+            # Update metadata with metadata from file
+            metadata.update(metadata_from_file)
```

Comment:
This is what happens when you spend too much time working with JavaScript 😆 . Much more Pythonic 🐍 .